.github/workflows/deployment.yml

```diff
@@ -42,49 +41,17 @@ jobs:
       domain: ${{ vars.DOMAIN_SUFFIX }}.${{ vars.DOMAIN }}
     steps:
 
-      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3
+      - name: Determine correct SHA to deploy
+        uses: actions/github-script@v6
+        id: determine-sha
+        with:
+          result-encoding: string
+          script: |
+            return (context.payload.action === 'ci-passed' ? context.payload.client_payload.sha : context.sha)
 
-      - name: log environment
-        id: log-environment
-        run: |
-          REF=$(git rev-parse HEAD)
-          echo "Git SHA: $REF"
-          echo "sha=$REF" | tr -d "\n" >> $GITHUB_OUTPUT
-
-          echo "github:"
-          <<-HEREDOC
-            ${{ toJSON(github) }}
-          HEREDOC
-          echo ""
-          echo "inputs:"
-          <<-HEREDOC
```

Comment:
I will add this again when i make the workflow repository independent.
Its very useful to detect newlines in environment variables and secrets which break the deployment

Reply:
For me it didn't echo anything, so I might have used it wrongly

Reply:
no, the cat was forgotten on most debug outputs. as i said, i will add it again, no problem

Reply:
Ah I see, thanks